### PR TITLE
arm64: support NVDIMM

### PR DESCRIPTION
**Pull request by Pennyzct**
As Arm64 is using the block device as the rootfs in guest. If we run two or more kata-container instances
on one host, we will get following error:
```
root@entos-thunderx2-desktop:~# docker run -dt --runtime kata-runtime ubuntu
7baf2c0f26100b0e642ad4122ce9ea1cb556fb3ed6cfcb454cb0586cbbe6194d
root@entos-thunderx2-desktop:~# docker run -dt --runtime kata-runtime ubuntu
2065fcf560c136091b5e69b6a12c95de94308309bb7fd5309df852503752203a
docker: Error response from daemon: OCI runtime create failed: qemu-system-aarch64: -device virtio-blk,drive=image-9f100592ac95eec6,scsi=off,config-wce=off: Failed to get "write" lock
Is another process using the image?: unknown.
```
Thanks to the new expanded IPA_SIZE feature in kernel 4.20 and Eric Auger's patch set [ARM virt: Initial RAM expansion and PCDIMM/NVDIMM support](https://www.mail-archive.com/qemu-devel@nongnu.org/msg599724.html) in qemu(which are still under upstream review), we could fully support nvdimm on arm64.

### CODE REVIEW

**Comment by Pennyzct** [comment]
I'm not triggering ARM CI here, since this pr needs to wait [packaging/#377](https://github.com/kata-containers/packaging/pull/377) landed firstly.

File: virtcontainers/qemu_arm64.go [hunk 1]
```diff
@@ -160,3 +163,29 @@ func (q *qemuArm64) bridges(number uint32) []types.PCIBridge {
 func (q *qemuArm64) appendBridges(devices []govmmQemu.Device, bridges []types.PCIBridge) []govmmQemu.Device {
 	return genericAppendBridges(devices, bridges, q.machineType)
 }
+
+func (q *qemuArm64) appendImage(devices []govmmQemu.Device, path string) ([]govmmQemu.Device, error) {
+	imageFile, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = imageFile.Close() }()
```
Comment (jodh-intel) [inline virtcontainers/qemu_arm64.go]: Nit: You could make this simply:

```go
defer imageFile.Close()
```

Comment (Pennyzct) [reply virtcontainers/qemu_arm64.go]: rignt, right, right. ;) update asap.

**Comment by Pennyzct** [comment]
@jodh-intel @sboeuf already updated. ;) ptal

**Comment by Pennyzct** [comment]
Hi~ all. @grahamwhaley @jodh-intel @sboeuf a few sad news and one great news.
Great news is that ARM CI got green~~~
But I got dozens of `build timeout` in multiple x86_64 CIs, and it seems like a pattern to me. you may want to have a look. ;)
```
•Build timed out (after 5 minutes). Marking the build as aborted.
Build was aborted
Performing Post build task...
Match found for :.* : True
Logical operation result is TRUE
Running script  : #!/bin/bash
```

**Comment by grahamwhaley** [comment]
Hi @Pennyzct \o/ for the ARM CI :-)
The timeouts - yes, I think we have had a bunch of those overnight - I think it may be related to an image file/clearlinux update that happened - I think @chavafg and @jcvenegas are looking into it. Thanks for reporting!

**Comment by grahamwhaley** [comment]
/test
This looks to be very ARM specific, and the ARM CI is happy. We know the x86 ones had some wobbles overnight - I'm probably happy to merge this as is. @sboeuf wdyt?
I'll kick the CIs off again anyway, as the 'wobbles' should be fixed now - but we could force merge without waiting if we wanted...

**Comment by sboeuf** [comment]
@grahamwhaley I agree this is ARM specific, but the code being modified is in virtcontainers. There could be some unexpected relation with the common code that could cause some failures. I would feel more confident to merge this if we had at least our main `jenkins-ci-ubuntu-18-04` job passing.

**Comment by jodh-intel** [comment]
Fedora CI failed with:

```
not ok 32 ctr with non-root user has no effective capabilities
```

Fedora vsocks CI failed with:

```
docker: Error response from daemon: OCI runtime create failed: Failed to check if grpc server is working: context deadline exceeded: unknown.
And we have hit the max 20 containers
Checking 20 containers have all relevant components
Wrong number of shims running (20 != 19) - stopping
Wrong number of netmons running (20 != 19) - stopping
Wrong number of 'runtime list' containers running (20 != 19) - stopping
Wrong number of pods in /var/lib/vc/sbs (20 != 19) - stopping)
Showing system state:
```

Let's re-spin the wheels and see if the errors are consistent...

**Comment by jodh-intel** [comment]
2 failing CI's are known issues, so merging...

**Comment by Pennyzct** [comment]
Hi~@jodh-intel this pr is tested on ARM CI with the dependency of [packaging#377](https://github.com/kata-containers/packaging/pull/377), for now, [packaging#377](https://github.com/kata-containers/packaging/pull/377) hasn't been merged yet, I'm afraid ARM ci could break down.
But I see that [packaging#377](https://github.com/kata-containers/packaging/pull/377) has got all green, maybe could we merge it?? ;)

**Comment by jodh-intel** [comment]
@Pennyzct - done ;)

**Pull request merged**